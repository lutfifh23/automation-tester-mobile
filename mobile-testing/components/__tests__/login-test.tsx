// components/__tests__/login-test.tsx
import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../../app/login'; // Sesuaikan dengan path yang benar

describe('Login Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
