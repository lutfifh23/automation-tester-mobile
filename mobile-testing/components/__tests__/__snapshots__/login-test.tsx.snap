// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "justifyContent": "center",
      "padding": 16,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 24,
        "marginBottom": 16,
      }
    }
    testID="title"
  >
    Login
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Username"
    style={
      {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 12,
        "paddingHorizontal": 8,
      }
    }
    testID="usernameInput"
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 12,
        "paddingHorizontal": 8,
      }
    }
    testID="passwordInput"
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="loginButton"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
